File: log-noop/src/main/java/eu/alessiobianchi/log/Log.java
package eu.alessiobianchi.log;

import java.io.File;

public final class Log {

	public static boolean isEnabled() {
		return false;
	}

	public static void setEnabled(boolean enabled) {
	}

	public static void init(boolean enableLogcat, File logFile) {
	}

	public static File getLogFile() {
		return null;
	}

	public static void v(String msg, Throwable t, Object tag) {
	}

	public static void d(String msg, Throwable t, Object tag) {
	}

	public static void i(String msg, Throwable t, Object tag) {
	}

	public static void w(String msg, Throwable t, Object tag) {
	}

	public static void e(String msg, Throwable t, Object tag) {
	}

	public static void v(String msg, Object tag) {
	}

	public static void d(String msg, Object tag) {
	}

	public static void i(String msg, Object tag) {
	}

	public static void w(String msg, Object tag) {
	}

	public static void e(String msg, Object tag) {
	}

}

File: log-noop/src/main/java/eu/alessiobianchi/log/Log.kt
@file:JvmName("Log")

package eu.alessiobianchi.log

import java.io.File
import java.util.concurrent.locks.ReentrantLock

object Log {
	@JvmStatic
	val lock = ReentrantLock()

	@JvmStatic
	var enabled = true
		@JvmName("isEnabled") get

	@JvmStatic
	var logFile: File? = null
		private set

	@JvmStatic
	fun init(enableLogcat: Boolean, logFile: File?) {
	}

	@JvmStatic
	inline fun withLock(crossinline block: () -> Unit) {
	}

	@JvmStatic
	@JvmOverloads
	fun v(msg: String?, t: Throwable?, tag: Any? = null) {
	}

	@JvmStatic
	@JvmOverloads
	fun d(msg: String?, t: Throwable?, tag: Any? = null) {
	}

	@JvmStatic
	@JvmOverloads
	fun i(msg: String?, t: Throwable?, tag: Any? = null) {
	}

	@JvmStatic
	@JvmOverloads
	fun w(msg: String?, t: Throwable?, tag: Any? = null) {
	}

	@JvmStatic
	@JvmOverloads
	fun e(msg: String?, t: Throwable?, tag: Any? = null) {
	}


	@JvmStatic
	@JvmOverloads
	fun v(msg: String?, tag: Any? = null) {
	}

	@JvmStatic
	@JvmOverloads
	fun d(msg: String?, tag: Any? = null) {
	}

	@JvmStatic
	@JvmOverloads
	fun i(msg: String?, tag: Any? = null) {
	}

	@JvmStatic
	@JvmOverloads
	fun w(msg: String?, tag: Any? = null) {
	}

	@JvmStatic
	@JvmOverloads
	fun e(msg: String?, tag: Any? = null) {
	}

	@JvmStatic
	fun getTag(obj: Any?): String = ""
}
